src/pybind/mgr/dashboard/frontend/src/app/ceph/new-dashboard/dashboard/dashboard.component.ts

```diff
@@ -23,13 +23,15 @@ import {
 } from '~/app/shared/services/feature-toggles.service';
 import { RefreshIntervalService } from '~/app/shared/services/refresh-interval.service';
 import { SummaryService } from '~/app/shared/services/summary.service';
+import { PrometheusListHelper } from '~/app/shared/helpers/prometheus-list-helper';
+import { PrometheusAlertService } from '~/app/shared/services/prometheus-alert.service';
 
 @Component({
   selector: 'cd-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.scss']
 })
-export class DashboardComponent implements OnInit, OnDestroy {
+export class DashboardComponent extends PrometheusListHelper implements OnInit, OnDestroy {
```

Comment:
First of all, I really love when I see some OOP stuff going on. That said, `DashboardComponent` is an Angular Component, and `PrometheusHelper` is an Angular Directive (with no selector, so it can't be used as Directive either) and which inherits from `ListWithDetails`, which in turn brings properties like `expandedRow` or `tableStatus`...

I think this is a bit messy inheritance model...

What do you want from `PrometheusHelper`? Some way to run code based on the online status of a backend component... For that, in some places we're using Status Guards (hooks in the Angular routing lifecycle to enable/activate pages based on conditions). However, that doesn't fit here.

So part of the issue is that we have to consume 2 service methods:
* `IsPrometheusConfigured()`: this offloads the internal state of the service to the consumer component, which is not polite (consumers shouldn't have to worry about the internal state of anything else that themselves')
* `getPrometheusData()`

If a devel just want to fetch Prometheus data, why forcing them to first check if Prometheus is configured? Let's follow the Python "better ask for forgiveness..." and detect that condition, either during the service instantiation or during the service method:

```ts
@Injectable({
  providedIn: 'root'
})
export class PrometheusService {
  private baseURL = 'api/prometheus';
  private settingsKey = { 
    alertmanager: 'api/settings/alertmanager-api-host',
    prometheus: 'api/settings/prometheus-api-host'
  };
  
  isPrometheusConfigured = false;  

  constructor(private http: HttpClient, private settingsService: SettingsService) 
  {
      this.isPrometheusConfigured$ = this.prometheusService.ifPrometheusConfigured().share();
  }

  getPrometheusData(params: any): any {
    return concat(
      this.isPrometheusConfigured$,
      this.http.get<any>(`${this.baseURL}/data`, { params })
    );
  }

```

Reply:
BTW the example above is not 100% functiona, the 2nd subscription still needs to check for the Prometheus configured status.

Reply:
the `prometheus-list.helper` was already there but it was only accessible to prometheus component. So what i did is to move that component to shared. I'll refactor this code, and it'll be affecting several other components. i'll check it.

Reply:
@epuertat for some days I've been trying this and i couldn't fully convert the existing behaviour. The original `ifPrometheusConfigured()` function doesn't returns a subscribable object. but even after making the changes that is needed to make it subscribable it doesn't work as it is before.